### ISSUE TITLE
Running with py39 scipy1.8, got noise in inpainting area from test set

### ISSUE THREAD

**Issue by yuguang-li** [ericsujw/LGPN-net#3]
I'm running the test code with Structure3D data set. My env setup is py39 and scipy 1.8. I replaced all imread(, mode='P') (deprecated in scipy) with `np.array(Image.open(path_to_img).convert('P'))`. The code runs, but produce something like below. I wonder if you can give me some clue on the potential cause.
![scene_03250_2D_rendering_861576_panorama](https://github.com/ericsujw/LGPN-net/assets/13266507/2f79a580-dd68-44f4-a726-e484ba923070)

**Comment by chencchengh** [comment]
Palette mode is only used when reading the semantic maps (used to generate furniture masks), so the noise is unlikely due to the modification.

**Comment by chencchengh** [comment]
The issue is more likely due to the wrong checkpoint file specified.

**Comment by yuguang-li** [comment]
I suspect about the checkpoint file too. Right now I'm using `InpaintingModel_gen_0.pth` from `checkpoints/train`. Do you know if that's the correct weight file from the correct epoch?

**Comment by chencchengh** [comment]
Can you provide your command for testing ?

**Comment by chencchengh** [comment]
And yes this should be the correct weight file.

**Comment by yuguang-li** [comment]
I tested on the folder from the download link. Below is my command `python test.py --checkpoint checkpoints/ --input gt/ --mask mask/ --output ./output`

**Comment by yuguang-li** [comment]
This is the line of dataset file I modified to. I didn't see any file with name `full_gt` or `empty_semantic`
```
        if self.config.MASK == 8 or self.config.MASK == 7 or self.config.FURNISHED == 1:
            ### load empty/full room semantic segmentation
            imgh, imgw = empty.shape[0:2]
            if self.training:
                empty_semantic_map = np.array(Image.open(os.path.join(self.structure3D_path, self.data[index], 'empty/semantic.png')).convert('P'))
                full_semantic_map = np.array(Image.open(os.path.join(self.structure3D_path, self.data[index], 'full/semantic.png')).convert('P'))
                # empty_semantic_map = imread(os.path.join(self.structure3D_path, self.data[index], 'empty/semantic.png'), mode='P')
                # full_semantic_map = imread(os.path.join(self.structure3D_path, self.data[index], 'full/semantic.png'), mode='P')
            else:
                empty_semantic_map = np.array(Image.open(self.data[index].replace('full_gt', 'empty_semantic')).convert('P'))
                full_semantic_map = np.array(Image.open(self.data[index].replace('full_gt', 'full_semantic')).convert('P'))
                # empty_semantic_map = imread(self.data[index].replace('full_gt', 'empty_semantic'), mode='P')
                # full_semantic_map = imread(self.data[index].replace('full_gt', 'full_semantic'), mode='P')
                
            empty_semantic_map = cv2.resize(empty_semantic_map.astype(np.uint8), (imgw, imgh), cv2.INTER_NEAREST)
            full_semantic_map = cv2.resize(full_semantic_map.astype(np.uint8), (imgw, imgh), cv2.INTER_NEAREST)
```

**Comment by chencchengh** [comment]
You have to specify --checkpoint checkpoints/train in order to get to the correct config file and checkpoint (suppose the checkpoint file is in the train/ subfolder). The full_gt and empty_semantic are the semantic segmentation files in the Structured3D dataset.

**Comment by yuguang-li** [comment]
It works now after I change the weight files. I actually didn't point to the semantic file of Structure3D and the results look reasonable. Is it normal as I imagine HorizonNet is doing the semantics?

**Comment by yuguang-li** [comment]
One last question, to re-train the model, is there any change I should do to the config?

**Comment by chencchengh** [comment]
Yes, HorizonNet is doing the semantics. If you are going to re-train the model on another dataset such as Zillow, you might want to modify dataset.py as some file names are hard coded. You might also want to tweak the mask setting as your new dataset might not contain instance segmentations like structured3D does. For re-training on Structured3D, change self.structure3D_path in dataset.py and you're probably good to go.

**Comment by yuguang-li** [comment]
Thanks!
